tests/catlistdisplayer/test-getPostTitle.php

```diff
@@ -0,0 +1,277 @@
+      self::$displayer_default->get_post_title(self::$test_post, 'h2')
+    );
+  }
+
+  public function test_class_without_tag() {
+
+    $displayer = new CatListDisplayer(
+      array_merge(self::$atts, ['title_class' => 'my-class'])
+    );
+
+    $expected = '<a href="' . get_permalink(self::$test_post) .'"'
+                . ' class="my-class"'
+                . ' title="Lcp test post">Lcp test post</a>';
+
+    // Shortcode param.
+    $this->assertSame(
+      $expected,
+      $displayer->get_post_title(self::$test_post)
+    );
+    // Function argument.
+    $this->assertSame(
+      $expected,
+      self::$displayer_default->get_post_title(self::$test_post, null, 'my-class')
+    );
+  }
+
+  public function test_class_and_tag() {
+
+    $displayer = new CatListDisplayer(
+      array_merge(
+        self::$atts,
+        ['title_class' => 'my-class', 'title_tag' => 'h2']
+      )
+    );
+
+    $expected = '<h2 class="my-class">'
+                . '<a href="' . get_permalink(self::$test_post) .'"'
+                . ' title="Lcp test post">Lcp test post</a></h2>';
+
+    // Shortcode params.
+    $this->assertSame(
+      $expected,
+      $displayer->get_post_title(self::$test_post)
+    );
+    // Function arguments.
+    $this->assertSame(
+      $expected,
+      self::$displayer_default->get_post_title(
+        self::$test_post, 'h2', 'my-class'
+      )
+    );
+  }
+
+  public function test_no_link_title() {
+
+    $displayer = new CatListDisplayer(
+      array_merge(self::$atts, ['link_titles' => 'false'])
+    );
+    $displayer2 = new CatListDisplayer(
+      array_merge(self::$atts, ['link_titles' => 'no'])
+    );
+
+    $this->assertSame(
+      'Lcp test post',
+      $displayer->get_post_title(self::$test_post)
+    );
+    $this->assertSame(
+      'Lcp test post',
+      $displayer2->get_post_title(self::$test_post)
+    );
+    $this->assertSame(
+      'Lcp test post',
+      self::$displayer_default->get_post_title(
+        self::$test_post, null, null, false
+      )
+    );
+  }
+
+  public function test_no_link_title_with_tag() {
+
+    $displayer = new CatListDisplayer(
+      array_merge(self::$atts,
+        ['link_titles' => 'false', 'title_tag' => 'h2']
+      )
+    );
+
+    $this->assertSame(
+      '<h2>Lcp test post</h2>',
+      $displayer->get_post_title(self::$test_post)
+    );
+    $this->assertSame(
+      '<h2>Lcp test post</h2>',
+      self::$displayer_default->get_post_title(
+        self::$test_post, 'h2', null, false
+      )
+    );
+  }
+
+  public function test_no_link_title_with_class() {
+
+    $displayer = new CatListDisplayer(
+      array_merge(self::$atts,
+        ['link_titles' => 'false', 'title_class' => 'my-class']
+      )
+    );
+
+    /*
+    How should the plugin behave when `link_titles=no` and `title_class=something` ?
+    Should it just ignore the class or should it add a default tag. If a default tag,
+    should it be <span>?
+     */
+  }
```

Comment:
@picandocodigo As you can see I've started fixing #350 by writing a test suite that `get_post_title` should pass but need to make a decision about this particular test case before writing assertions. What do you think?

Reply:
@zymeth25 I think using `span` makes sense :+1:

Reply:
Yeah, sounds good.